databricks/koalas/frame.py

```diff
@@ -1283,6 +1283,41 @@ def notnull(self):
 
     notna = notnull
 
+    def nunique(self, axis=0, dropna=True):
+        """
+        Return number of unique elements in the object.
+
+        Excludes NA values by default.
+
+        Parameters
+        ----------
+        axis : Can only be set to 0 at the moment.
+        dropna : Don’t include NaN in the count.
+
+        Returns
+        -------
+        The number of unique values as an int.
+
+        Examples
+        --------
+        >>> ks.DataFrame({'A': [1, 2, 3], 'B': [np.nan, 3, np.nan]}).nunique()
+        A    3
+        B    1
+        Name: 0, dtype: int64
+
+        >>> ks.DataFrame({'A': [1, 2, 3], 'B': [np.nan, 3, np.nan]}).nunique(dropna=False)
+        A    3
+        B    2
+        Name: 0, dtype: int64
+        """
+        if axis != 0:
+            raise ValueError("The 'nunique' method only works with axis=0 at the moment")
+        res = self._sdf.select(*(F.countDistinct(Column(c)).alias(c)for c in self.columns))
+        res = res.toPandas().T.iloc[:, 0]
+        if not dropna:
+            res += self.isnull().sum().clip(upper=1)
```

Comment:
It'd be better if we can do this in a single pass, rather than triggering two jobs. Take a look at the current is_unique property and you can transfer that "algorithm" here.

Reply:
I've fixed this now, so with 0f5b16b it's only a single job.